### PR TITLE
Add documentation about `pyproject.toml`

### LINKED ISSUE #2377: Add/improve documentation about `pyproject.toml` (first draft by development team)
## Description
I don't feel like I have any grasp of what `pyproject.toml` is, partly because i'm not a python dev. But I'm seeing it mentioned a fair bit (and we have a page on settings that mentions it in comparison to runtime settings and setup.py). I think it would be a good idea to trawl through the docs and find out what we have and create a section somewhere that is comprehensive/complete as a reference to what it is, how to update it, what we include in it etc.

cc @astrojuanlu as the main inspiration for this, having reviewed my PR for #1985 and spotted the inconsistency.

### PR DESCRIPTION
**Pull request by ankatiyar**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Resolves #2377 
## Development notes

I've added a section about the settings contained in `pyproject.toml` to the "Project Settings" (`docs/source/kedro_project_setup/settings.md`) page in "Kedro Project Setup" section 
- the link I removed in the `docs/source/get_started/kedro_concepts.md` isn't rendered as a link 
- updated `project_version` to `kedro_init_version` in the faqs
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by datajoely** [comment]
`kedro_init_version` is a much better name

**Comment by datajoely** [comment]
image

I made this a while ago when were designing `settings.py` which is a Django inspired concept IIRC

**Comment by astrojuanlu** [comment]
Let's continue the conversation about terminology in https://github.com/kedro-org/kedro/issues/2434.

**Comment by astrojuanlu** [comment]
RTD build because of:

```
/home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/2427/docs/kedro_project_setup/settings.md:8: WARNING: 'myst' reference target not found: #settings-in-pyproject.toml
/home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/2427/docs/kedro_project_setup/settings.md:29: WARNING: Could not lex literal_block as "toml". Highlighting skipped.
```

File: docs/source/kedro_project_setup/settings.md [hunk 1]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
 | `CONFIG_LOADER_CLASS`       | `kedro.config.ConfigLoader`                       | Customise how project configuration is handled.                                                                    |
 | `CONFIG_LOADER_ARGS`        | `dict()`                                          | Keyword arguments for the `CONFIG_LOADER_CLASS` constructor.                                                       |
 | `DATA_CATALOG_CLASS`        | `kedro.io.DataCatalog`                            | Customise how the [Data Catalog](../data/data_catalog.md) is handled.                                              |
+
+## Settings in `pyproject.toml`
+The `pyproject.toml` file is defined in [PEP 518](https://peps.python.org/pep-0518/) as a configuration file to store build system requirements for Python projects.
+The `pyproject.toml` file in your project contains the metadata for your project and other tool specific configurations.
```
Comment (astrojuanlu) [inline docs/source/kedro_project_setup/settings.md:27]: Historically yes, but later on it has been extended by PEP 517, and most importantly PEP 621. In addition, many tools started using it to store their settings (Kedro among them, but also most of the ecosystem, including linters etc). Not sure if we should get into specifics here (this is mostly interesting for Python packaging nerds like myself):

```suggestion
The `pyproject.toml` file is the standard way to store build metadata and tool settings for Python projects.
```

File: docs/source/kedro_project_setup/settings.md [hunk 2]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+Every Kedro project comes with a default pre-populated `pyproject.toml` file in your project root directory with the following keys specified under the `[tool.kedro]` section :
+```toml
+[tool.kedro]
+package_name = <package_name>
+project_name = <project_name>
+kedro_init_version = <kedro_version>
+```
+
+The `package_name` which should contain a [valid Python package name](https://peps.python.org/pep-0423/) for your project and `project_name` which should be a human-readable name for your project are mandatory keys.
```
Comment (astrojuanlu) [inline docs/source/kedro_project_setup/settings.md:36]: I'm looking forward for us to merge `setup.py` and `pyproject.toml` so we can unify parts of this...

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Added one suggestion about the wording, but other than that it's great having this finally, thanks @ankatiyar! 🙌🏽

File: docs/source/kedro_project_setup/settings.md [hunk 3]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
 | `CONFIG_LOADER_CLASS`       | `kedro.config.ConfigLoader`                       | Customise how project configuration is handled.                                                                    |
 | `CONFIG_LOADER_ARGS`        | `dict()`                                          | Keyword arguments for the `CONFIG_LOADER_CLASS` constructor.                                                       |
 | `DATA_CATALOG_CLASS`        | `kedro.io.DataCatalog`                            | Customise how the [Data Catalog](../data/data_catalog.md) is handled.                                              |
+
+## Settings in `pyproject.toml`
+The `pyproject.toml` file is defined in [PEP 518](https://peps.python.org/pep-0518/) as a configuration file to store build system requirements for Python projects.
+The `pyproject.toml` file in your project contains the metadata for your project and other tool specific configurations.
+Every Kedro project comes with a default pre-populated `pyproject.toml` file in your project root directory with the following keys specified under the `[tool.kedro]` section :
```
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:28]:
```suggestion
Every Kedro project comes with a default pre-populated `pyproject.toml` file in your project root directory with the following keys specified under the `[tool.kedro]` section:

```

File: docs/source/kedro_project_setup/settings.md (hunk 2 above)
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:36]:
```suggestion
The `package_name` should be a [valid Python package name](https://peps.python.org/pep-0423/) and the `project_name` should be a human-readable name. They are both mandatory keys for your project.
```

File: docs/source/kedro_project_setup/settings.md [hunk 4]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+```toml
+[tool.kedro]
+package_name = <package_name>
+project_name = <project_name>
+kedro_init_version = <kedro_version>
+```
+
+The `package_name` which should contain a [valid Python package name](https://peps.python.org/pep-0423/) for your project and `project_name` which should be a human-readable name for your project are mandatory keys.
+`kedro_init_version` specifies the version of Kedro the project was created with. When you upgrade to a newer Kedro version,
+this value should also be updated. <br>
```
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:38]:
```suggestion
this value should also be updated.


```

File: docs/source/kedro_project_setup/settings.md [hunk 5]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+You can also use `pyproject.toml` to specify settings for functionalities such as [micro-packaging](../nodes_and_pipelines/micro_packaging.md)
```
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:39]:
```suggestion
You can also use `pyproject.toml` to specify settings for functionalities such as [micro-packaging](../nodes_and_pipelines/micro_packaging.md).
```

File: docs/source/kedro_project_setup/settings.md [hunk 6]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+Additionally, you can also store the settings for the other tools you've used in your project, such as [`pytest` for automated testing](../development/automated_testing.md).
```
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:40]:
```suggestion
You can also store the settings for the other tools you've used in your project, such as [`pytest` for automated testing](../development/automated_testing.md).
```

File: docs/source/kedro_project_setup/settings.md [hunk 7]
```diff
@@ -19,3 +21,21 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+You can consult the respective documentation of the tools you have used to check how you can configure the settings using the `pyproject.toml` file for your project. <br>
```
Comment (stichbury) [inline docs/source/kedro_project_setup/settings.md:41]:
```suggestion
Consult the respective documentation for the tools you have used to check how you can configure the settings with the `pyproject.toml` file for your project. 
```

**Review by stichbury — APPROVED** [review]
Nice work -- thank you! Just a few minor tweaks 🏆 🏅

File: docs/source/faq/faq.md [hunk 8]
```diff
@@ -132,7 +132,7 @@ To later upgrade Kedro to a different version, simply run:
 pip install kedro -U
 ```
 
-When migrating an existing project to a newer Kedro version, make sure you also update the `project_version` in your `pyproject.toml` file from the project root directory or, for projects generated with Kedro<0.17.0, in your `ProjectContext`, which is found in `src/<package_name>/run.py`.
+When migrating an existing project to a newer Kedro version, make sure you also update the `kedro_init_version` in your `pyproject.toml` file from the project root directory or, for projects generated with Kedro<0.17.0, in your `ProjectContext`, which is found in `src/<package_name>/run.py`.
```
Comment (ankatiyar) [inline docs/source/faq/faq.md:136]: This change is not really needed after #2373 is merged since this part is being removed.

Comment (stichbury) [reply docs/source/faq/faq.md:136]: Thanks! The section is moving to the installation guide, so I have made the change as part of that PR 🙏

**Comment by stichbury** [comment]
@ankatiyar Just to say that I fixed a couple of merge issues as a result of my FAQ PR, which is now in `main`.

**Comment by ankatiyar** [comment]
> @ankatiyar Just to say that I fixed a couple of merge issues as a result of my FAQ PR, which is now in `main`.

Thanks @stichbury! 👍

File: docs/source/kedro_project_setup/settings.md [hunk 9]
```diff
@@ -19,3 +21,22 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+kedro_init_version = kedro_version
+```
+
+The `package_name` should be a [valid Python package name](https://peps.python.org/pep-0423/) and the `project_name` should be a human-readable name. They are both mandatory keys for your project.
+`kedro_init_version` specifies the version of Kedro the project was created with. When you upgrade to a newer Kedro version,
+this value should also be updated.
+
+You can also use `pyproject.toml` to specify settings for functionalities such as [micro-packaging](../nodes_and_pipelines/micro_packaging.md).
+You can also store the settings for the other tools you've used in your project, such as [`pytest` for automated testing](../development/automated_testing.md).
+Consult the respective documentation for the tools you have used to check how you can configure the settings with the `pyproject.toml` file for your project.
```
Comment (noklam) [inline docs/source/kedro_project_setup/settings.md:42]: Just a wild idea, can we link our `pyproject.toml` as an example? 

https://github.com/kedro-org/kedro/blob/main/pyproject.toml

File: docs/source/kedro_project_setup/settings.md [hunk 10]
```diff
@@ -19,3 +21,22 @@ By default, all code in `settings.py` is commented out. When settings are not su
...
+```toml
+[tool.kedro]
+package_name = package_name
+project_name = project_name
+kedro_init_version = kedro_version
+```
+
+The `package_name` should be a [valid Python package name](https://peps.python.org/pep-0423/) and the `project_name` should be a human-readable name. They are both mandatory keys for your project.
+`kedro_init_version` specifies the version of Kedro the project was created with. When you upgrade to a newer Kedro version,
+this value should also be updated.
```
Comment (noklam) [inline docs/source/kedro_project_setup/settings.md:38]: Isn't it the version of the "starter"? Just double checking

Comment (ankatiyar) [reply docs/source/kedro_project_setup/settings.md:38]: I think it's the version of Kedro used to create the project. It's also there when you make a blank project with `kedro new`

**Pull request merged**